xtask/src/hooks.rs

```diff
@@ -0,0 +1,161 @@
+use std::fs::File;
+use std::io::{BufRead, BufReader, ErrorKind};
+use std::path::Path;
+use std::process::Command;
+
+use clap::Args;
+use cross::CommandExt;
+
+const CARGO_FLAGS: &[&str] = &["--all-features", "--all-targets", "--workspace"];
+
+#[derive(Args, Debug)]
+pub struct Check {
+    /// Provide verbose diagnostic output.
+    #[clap(short, long)]
+    verbose: bool,
+    /// Run shellcheck on all files, not just staged files.
+    #[clap(short, long)]
+    all: bool,
+}
+
+#[derive(Args, Debug)]
+pub struct Test {
+    /// Provide verbose diagnostic output.
+    #[clap(short, long)]
+    verbose: bool,
+}
+
+fn has_nightly(verbose: bool) -> cross::Result<bool> {
+    Command::new("cargo")
+        .arg("+nightly")
+        .run_and_get_output(verbose)
+        .map(|o| o.status.success())
+        .map_err(Into::into)
+}
```

Comment:
technically, this should be
```suggestion
fn has_nightly(verbose: bool) -> cross::Result<bool> {
    Command::new("cargo")
        .arg("+nightly")
        .run_and_get_output(verbose)
        .map(|o| o.stdout.contains("-nightly"))
        .map_err(Into::into)
}
```

but that's just being very pedantic

Reply:
Why though? I've just checked that running `cargo +nightly` produces a non-zero error code: if it works, it works. If someone wants to rename it something funny, then sure. If I do:

```bash
$ cargo +sofa
error: toolchain 'sofa' is not installed
```

Reply:
exactly, it's very uneccessary :)